formsfeeder.client/src/test/java/formsfeeder/client/FormsFeederClientTest.java

```diff
@@ -171,7 +176,46 @@ public void tearDown() throws Exception {
 	        wireMockServer.stop();
 		}
 	}
-	
+
+	@Test
+	void testAccept_HeadersAdded() throws Exception {
+		String correlationId = "correlationId";
+		String headerName1 = "headerName1";
+		String headerValue1= "headerValue1";
+		String headerName2 = "headerName2";
+		String headerValue2= "headerValue2";
+
+		FormsFeederClient underTest = FormsFeederClient.builder()
+				.machineName(formsfeederServerName)
+				.port(formsfeederServerPort)
+				.plugin("Debug")
+				.correlationId(()->correlationId)
+				.addHeader(headerName1,()->headerValue1)
+				.addHeader(headerName2,()->headerValue2)
+				.build();
+		DataSourceList result = underTest.accept(DataSourceList.emptyList());
+
+		wireMockServer.verify(getRequestedFor(urlMatching("/api/v1/Debug"))
+				.withHeader(headerName1,matching(headerValue1))
+				.withHeader(headerName2,matching(headerValue2))
+				.withHeader(CorrelationId.CORRELATION_ID_HDR,matching(correlationId)));
+	}
+
+	@Test
+	void testAccept_CustomContextRoot() throws Exception {
+		FormsFeederClient underTest = FormsFeederClient.builder()
+				.machineName(formsfeederServerName)
+				.port(formsfeederServerPort)
+				.plugin("Debug")
+				.contextRoot("/custom/context/root/")
+				.build();
+		DataSourceList result = underTest.accept(DataSourceList.emptyList());
+
+		wireMockServer.verify(getRequestedFor(urlMatching("/custom/context/root/Debug"))
+				.withHeader(CorrelationId.CORRELATION_ID_HDR,matching(underTest.returnedCorrelationId())));
+
+	}
+

```

Comment:
This looks great.  I haven't used WireMock's verification capabilities before so this is educational for me.  I have just one request.  I've been using these client tests for both unit testing and integration testing (for integration testing, I fire up a formsfeeder server locally and change the USE_WIREMOCK variable to false before running the tests - it executes the tests directly against the formsfeeder server).  

By rights, I should use a System property to enable integration testing instead of making a code change.  This would also allow us to enable/disable tests based on that System property.  In the absence of doing that, if you could make these tests conditional upon the USE_WIREMOCK variable.

This is what I am thinking:
```Java
	@Test
	void testAccept_CustomContextRoot() throws Exception {
		if (USE_WIREMOCK) {	// Perform this test when doing unit testing (using wiremock), but skip this test when doing integration testing
			FormsFeederClient underTest = FormsFeederClient.builder()
					.machineName(formsfeederServerName)
					.port(formsfeederServerPort)
					.plugin("Debug")
					.contextRoot("/custom/context/root/")
					.build();
			DataSourceList result = underTest.accept(DataSourceList.emptyList());
	
			wireMockServer.verify(getRequestedFor(urlMatching("/custom/context/root/Debug"))
					.withHeader(CorrelationId.CORRELATION_ID_HDR,matching(underTest.returnedCorrelationId())));
		}
	}
```

This will allow me to continue to use the same tests for unit and integration testing.

At some point in the future, I will move to using a System property and adding a JUnit annotation like this to the test:
`@DisabledIfSystemProperty(named = "com._4point.formsfeeder.test-type", matches = "integration")`

Reply:
I've never used WireMock so this whole thing has been learning for me :) Yeah I'll get that updated here shortly.